src/uwtools/utils/file.py

```diff
@@ -7,13 +7,62 @@
 import shutil
 import sys
 from contextlib import contextmanager
+from dataclasses import dataclass
 from datetime import datetime as dt
 from pathlib import Path
 from typing import IO, Generator
 
 from uwtools.types import DefinitePath, OptionalPath
 
 
+@dataclass(frozen=True)
+class _FORMAT:
+    """
+    A mapping from config format names to literal strings.
+    """
+
+    # Canonical strings:
+
+    _atparse: str = "atparse"
+    _fieldtable: str = "fieldtable"
+    _ini: str = "ini"
+    _jinja2: str = "jinja2"
+    _nml: str = "nml"
+    _yaml: str = "yaml"
+
+    # Variants:
+
+    atparse: str = _atparse
+    bash: str = _ini
+    cfg: str = _ini
+    fieldtable: str = _fieldtable
+    ini: str = _ini
+    jinja2: str = _jinja2
+    nml: str = _nml
+    sh: str = _ini
+    yaml: str = _yaml
+    yml: str = _yaml
+
+
+FORMAT = _FORMAT()
```

Comment:
Here's the dataclass object now accessed elsewhere to refer to specific strings used as config filename extensions and CLI values.